### PR TITLE
[8.0]-[FIX]-Issues BR 240,241,242

### PR DESCRIPTION
**Pull request by YogeshMahera-SerpentCS**
Fixed Issues

#240 #241 #242

### CODE REVIEW

**Comment by YogeshMahera-SerpentCS** [comment]
@victormartinelicocorp can you please review it.

File: business_requirement_deliverable_project/wizard/generate_projects.py [hunk 1]
```diff
@@ -151,7 +152,8 @@ def generate_deliverable_projects(
                 line_project_val = self._prepare_project_vals(
                     line, parent_project)
                 line_project_val.update({
-                    'business_requirement_deliverable_id': line.id
+                    'business_requirement_deliverable_id': line.id,
+                    'business_requirement_id': line.business_requirement_id.id
                 })
                 line_project = project_obj.create(line_project_val)
                 line.linked_project = line_project.id
```
Comment (martintorresvm) [inline business_requirement_deliverable_project/wizard/generate_projects.py]: @YogeshMahera-SerpentCS any reason to keep this field `linked_project`?

Comment (martintorresvm) [reply business_requirement_deliverable_project/wizard/generate_projects.py]: These two fields:

https://github.com/OCA/business-requirement/pull/245/files#diff-e9c58cb738de57ebabe8fec326544d7fR16
https://github.com/OCA/business-requirement/pull/245/files#diff-e9c58cb738de57ebabe8fec326544d7fR126

**Review by elicoidal — APPROVED** [review]
Functional tests are OK.
Please attend @victormartinelicocorp concerns

File: business_requirement_deliverable_project/wizard/generate_projects.py (hunk 1 above)
Comment (YogeshMahera-SerpentCS) [reply business_requirement_deliverable_project/wizard/generate_projects.py]: yes no need to keep,I had checked it's not usable so removed it.
@victormartinelicocorp

**Comment by YogeshMahera-SerpentCS** [comment]
@victormartinelicocorp could you please review

File: business_requirement_deliverable_project/wizard/generate_projects.py (hunk 1 above)
Comment (martintorresvm) [reply business_requirement_deliverable_project/wizard/generate_projects.py]: @YogeshMahera-SerpentCS please note that you haven't.

https://github.com/OCA/business-requirement/pull/245/files#diff-e9c58cb738de57ebabe8fec326544d7fR16
https://github.com/OCA/business-requirement/pull/245/files#diff-e9c58cb738de57ebabe8fec326544d7fR126

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal @victormartinelicocorp Could you please review it again.

**Comment by elicoidal** [comment]
@victormartinelicocorp @seb-elico can you review?

File: business_requirement_deliverable_project/models/business.py [hunk 2]
```diff
@@ -13,12 +13,10 @@ def generate_projects_wizard(self):
             br_ids=self).generate_project_wizard()
         return res
 
-    linked_project = fields.Many2one(
-        string='Linked project',
+    project_ids = fields.One2many(
         comodel_name='project.project',
-        groups='project.group_project_user',
```
Comment (martintorresvm) [inline business_requirement_deliverable_project/models/business.py]: @YogeshMahera-SerpentCS I didn't expect this change. Why is necessary remove these groups, readonly, copy, etc...?

File: business_requirement_deliverable_project/models/business.py [hunk 3]
```diff
@@ -43,12 +41,12 @@ def generate_projects_wizard(self):
         compute='_compute_planned_hour'
     )
     linked_project_count = fields.Integer(
-        compute='_compute_linked_project_count',
+        compute='_compute_projects_count',
```
Comment (martintorresvm) [inline business_requirement_deliverable_project/models/business.py]: @YogeshMahera-SerpentCS Please keep the name function.

File: business_requirement_deliverable_project/models/business.py [hunk 4]
```diff
@@ -122,9 +121,33 @@ def _compute_planned_hour(self):
 class BusinessRequirementDeliverable(models.Model):
     _inherit = "business.requirement.deliverable"
 
-    linked_project = fields.Many2one(
-        string='Linked project',
+    project_ids = fields.One2many(
         comodel_name='project.project',
-        groups='project.group_project_user',
```
Comment (martintorresvm) [inline business_requirement_deliverable_project/models/business.py]: @YogeshMahera-SerpentCS I just need an explanation about this.

**Review by martintorresvm — CHANGES_REQUESTED** [review]

**Review by martintorresvm — APPROVED** [review]
For the record we still could need a `migrations` folder with a procedure for let the users keep information.

**Comment by martintorresvm** [comment]
@elicoidal Please merge it

cc @seb-elico

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS please push the changes to v10 too.
thanks

**Pull request merged**